homeassistant/components/brother/config_flow.py

```diff
@@ -30,9 +30,9 @@ def host_valid(host: str) -> bool:
         if ipaddress.ip_address(host).version in [4, 6]:
             return True
     except ValueError:
-        disallowed = re.compile(r"[^a-zA-Z\d\-]")
-        return all(x and not disallowed.search(x) for x in host.split("."))
-    return False
+        pass
+    disallowed = re.compile(r"[^a-zA-Z\d\-]")
+    return all(x and not disallowed.search(x) for x in host.split("."))
```

Comment:
Is this a fix?

Reply:
This increases test coverage. Currently, line 35 has no coverage in the tests.

Reply:
Do we want the regex search to habdke the case that ip address version is not in 4 or 6?

Reply:
Regex handles the validation of the hostname.

Reply:
Right, but this change removes a third case where the ip address version is not 4 or 6. If that can't happen why do we have that check?

Reply:
Maybe I don't understand your question... if `host` is not IPv4 or v6 than we check if it is a valid hostname. We have tests for that and they passe with this change.
https://github.com/home-assistant/core/blob/dev/tests/components/brother/test_config_flow.py#L27-L40
https://github.com/home-assistant/core/blob/dev/tests/components/brother/test_config_flow.py#L79-L87

What third case do you mean? `host` is IPv4 or IPv6 or hostname or invalid.

Reply:
Ok, then it's good.

There are three cases in the flow:
1. no exception and ip4 or ip6
2. no exception and not ip4 or ip6
3. exception

My question was if we wanted to handle both 2 and 3 with the same check.

Reply:
Since this change is unrelated to the state class attributes it should have probably been a separate PR.

Reply:
OK, I'll move this to another PR.